### ISSUE TITLE
Merge tool and skeleton packages

### ISSUE THREAD

**Issue by psimk** [mediamonks/porter#14]
The current setup of having separated packages for build tools and skeleton specific tools is nice because it makes future project upgrading possible. Additionally, you can use multiple tools in the same project, making some previously difficult workflows possible.

However, this is confusing.

First, there is no clear definition on what should be included in a build tool package or a skeleton package. It is difficult to come up with clear definitions as there are always grey areas - should a linting tool be part of the skeleton or build tool package? It can be used either during development, bundling or committing changes. Depending on the skeleton, the requirements are different and we can't have a clear definition.

Secondly, it is difficult and annoying to maintain. If there are changes than span both the build tool and the skeleton, then there are two packages that have to be changes and updated. This increases the maintenance time and also raises the entry barrier for contributors. This is even more visible for when someone wants to create a completely new skeleton with a new tool for porter: you immediately have to setup 2 packages and either 2 repositories or get down with a mono-repo. Most developers won't want to go through all of this hassle.

In summary, the 2 package requirement is inefficient and I think that, while it makes porter and its skeleton config easier to manage, the complexity for skeleton developers is too high. From a conceptual point of view, explaining that you need 2 packages also makes a lot of developers confused, "isn't it just a skeleton?".

I think the correct route would be to merge the tool and skeleton packages and make porter capable of managing nested skeletons. For example, we can have a base webpack skeleton which the react skeleton uses as a dependency. The react skeleton could extend webpack skeleton's configuration and change things specific to React. This way, we don't loose the composability and upgradability of packages, everything would just be more streamlined.
